mlflow/pyspark/ml/__init__.py

```diff
@@ -0,0 +1,180 @@
+import json
+from pkg_resources import resource_filename
+
+import mlflow
+from mlflow.utils import chunk_dict, truncate_dict
+from mlflow.utils.annotations import experimental
+from mlflow.utils.autologging_utils import (
+    autologging_integration,
+    safe_patch,
+    try_mlflow_log,
+)
+
+from mlflow.spark import FLAVOR_NAME
+
+
+def _get_fully_class_name(instance):
+    return instance.__class__.__module__ + "." + instance.__class__.__name__
+
+
+def _read_log_model_allowlist():
+    """
+    Reads the module allowlist and returns it as a set of tuples.
+    """
+    file_path = resource_filename(__name__, "log_model_allowlist.txt")
+    allowlist = set()
+    with open(file_path) as f:
+        for line in f:
+            stripped = line.strip()
+            is_blankline_or_comment = stripped == "" or stripped.startswith("#")
+            if not is_blankline_or_comment:
+                allowlist.add(stripped)
+    return allowlist
+
+
+_log_model_allowlist = _read_log_model_allowlist()
+
+
+def _should_log_model(estimator, spark_model):
+    class_name = _get_fully_class_name(estimator)
+    if class_name in _log_model_allowlist:
+        return True
+    elif class_name == 'pyspark.ml.classification.OneVsRest':
+        classifier = estimator.getClassifier()
+        return _get_fully_class_name(classifier) in _log_model_allowlist
+    elif class_name == 'pyspark.ml.classification.LogisticRegression':
+        return spark_model.numClasses < 10
```

Comment:
Is this because the model can be large if the number of classes is large?

Reply:
yes.

Reply:
How did you decide 10 is the right threshold?